spring-cloud-schema-registry-core/src/main/java/org/springframework/cloud/schema/registry/model/Schema.java

```diff
@@ -46,8 +53,12 @@
 	@Column(name = "FORMAT", nullable = false)
 	private String format;
 
-	@Lob
-	@Column(name = "DEFINITION", nullable = false, length = 8192)
+	@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "subject")
+	@JsonIdentityReference(alwaysAsId = true)
+	@ManyToMany
+	private List<Schema> references = new ArrayList<>();
```

Comment:
Another issue I stumbled upon, since now instead of returning all `Schema` references, I only return their `subject` names i.e.:
```
{
    "subject": "test",
    "id": 1,
    "version": 1,
    "format": "avro",
    "references": [
        "subject.reference"
    ],
    "definition": "..."
}
```
as opposed to:
```
{
    "subject": "test",
    "id": 1,
    "version": 1,
    "format": "avro",
    "references": [
        {
                "subject": "subject.reference",
                "id": 2,
                 ...
        }
    ],
    "definition": "..."
}
```

This however, now breaking tests since `MappingJackson2HttpMessageConverter`'s object mapper is trying to map REST's response payload directly to `Schema.class` where `Schema.references` is `List<Schema>` not `List<String>`.

I am not entirely sure how to solve this for now.

1. Write an interceptor to fetch referenced schemas during serialization
2. Map REST's response payload to `String`, then perform manual parsing
3. Somehow deactivate `@Json*` annotations under `Schema.class` during tests, so that the latter payload format gets returned